service/src/test/java/uk/gov/hmcts/reform/fpl/service/OrganisationServiceTest.java

```diff
@@ -138,4 +141,50 @@ void shouldRethrowExceptionOtherThanNotFound() {
             () -> organisationService.findUserByEmail(USER_EMAIL));
         assertThat(actualException).isEqualTo(exception);
     }
+
+    @Test
+    void shouldFindOrganisation() {
+        when(organisationApi.findOrganisationById(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID))
+            .thenReturn(buildOrganisation());
+
+        Organisation actualOrganisation = organisationService.findOrganisation();
+
+        assertThat(actualOrganisation).isEqualTo(buildOrganisation());
+    }
+
+    @Test
+    void shouldReturnEmptyOrganisationBuilderWhenOrganisationNotFound() {
+        when(organisationApi.findOrganisationById(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID))
+            .thenThrow(new FeignException.NotFound("Organisation not found", REQUEST, new byte[]{}));
+
+        Organisation organisation = organisationService.findOrganisation();
+
+        assertThat(organisation).isEqualTo(Organisation.builder().build());
+    }
+
+    @Test
+    void shouldThrowFeignExceptionWhenOrganisationIsNotFound() {
+        when(organisationApi.findOrganisationById(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID))
+            .thenThrow(new FeignException.NotFound("Organisation not found", REQUEST, new byte[]{}));
+
+        assertThatThrownBy(() -> organisationApi.findOrganisationById(AUTH_TOKEN_ID, SERVICE_AUTH_TOKEN_ID))
+            .isInstanceOf(FeignException.NotFound.class);
+    }
+
+    private Organisation buildOrganisation() {
+        return Organisation.builder()
+            .name("Organisation")
+            .contactInformation(buildOrganisationContactInformation())
+            .build();
+    }
+
+    private List<ContactInformation> buildOrganisationContactInformation() {
+        return List.of(ContactInformation.builder()
+            .addressLine1("Flat 12, Pinnacle Apartments")
+            .addressLine1("Saffron Central")
+            .county("London")
+            .country("United Kingdom")
+            .postCode("CR0 2GE")
+            .build());
+    }
```

Comment:
in this class I would do something like set Organisation.builder().build() and buildOrganisation() to global variables. EMPTY and POPULATED or something like that. What do you think?

Reply:
happy with this! done